RIPS/rip-7560.md

```diff
@@ -344,6 +344,42 @@ The gas limit of this frame is set to `paymasterPostOpGasLimit`.
 Revert in the `postPaymasterTransaction` frame reverts the transaction's execution frame as well.
 The gas fees charged from the `paymaster` will include the validation frames and also gas cost of the reverted execution frame and postPaymasterTransaction frame.
 
+### System Transaction Events
+
+We define the following system-level events that are emitted as part of an RIP-7560 transaction:
+
+```solidity
+    event RIP7560TransactionEvent(
+        address indexed sender,
+        address indexed paymaster,
+        address indexed deployer,
+        uint256 nonce,
+        bool success,
+        uint256 actualGasCost,
+        uint256 actualGasUsed
+    );
+
+    event RIP7560TransactionRevertReason(
```

Comment:
revert events provide little information - and miss the most common `revert("")` (which is indistinguishable from "OOG") - and of course, miss the "root cause" contract, as they only report top-level revert.

applications must use `callTracer` to gain such insight, so there is no real merit for using events.

analytics tools such as "dune" might gain some info from these events (e.g. the existing 4337 "front-running" reporting)
However, these tool ALREADY support tracing to provide deeper insights, so such scripts are still possible (even though are little more complex)

Reply:
We can trivially add `bool outOfGas` parameter to the `RevertReason` event. I did not add it because there is no such field in ERC-4337. Please advice.

Reply:
That's not the point: I didn't mean "OOG is the only missing revert", but that revert reasons give little information. They give little more info that "reverted", but for real troubleshooting, you must do callTrace somehow.